docs/package-deletion-policies.md

```diff
@@ -0,0 +1,210 @@
+
+What if a package (or version series) is no longer in active development?
+What if a specific version has a security issue?
+In these cases, the community should be informed of the issues, and possibly prevented from using a specific version, but deleting the package outright is not a tenable solution.
+So what can package managers do in these cases?
+
+### Yanking
+
+"Yanking" refers to soft deleting a compromised package version.
+The version still exists, but by marking the version as yanked, installers will ignore the version unless it is the only release that matches a version specifier.
+If the yanked version is explicitly required, it will be installed and will not break the build.
+Maintainers should consider yanking a release if it is broken, unstable, or contains a very severe vulnerability.
+Yanking will not completely guarantee that no one is using the yanked version, but it will reduce the installer preference to install it.
+If an ecosystem allows for version yanking, it is best practice to also allow maintainers to provide a reason.
+Users who are explicitly requiring the yanked version can then appropriately prioritize moving to a non-yanked version.
+
+Some version yanking policies use slightly different terminology but are functionally very similar.
+For example, Go uses the term "retract" and the retract directive can indicate that a package version (or version range) should not be used.
+
+Regardless of the term used, the following sets these policies apart from a deletion policy in the following ways:
+
+1. The yanked version is still available.
+2. The installer will try to avoid using yanked versions.
+3. The package consumer can make an informed decision about whether to use the yanked version.
+4. Yanking a version will not break a build.
+
+### Deprecation
+
+Software projects are not supported forever.
+When the maintainer/maintainer team is no longer willing to support a specific version series or an entire project, the team can mark the project as deprecated.
+Generally, deprecated projects are available for installation and can be continued to be used.
+Marking a project as deprecated allows users to make an informed decision about the use of the package.
+Going forward, the user knows that not only can they not expect any new features, but they can't even expect continued compatibility with other projects or security updates.
+
+Similar to version yanking policies, deprecation policies could use slightly different languages including terms like 'end of life' or mention moving to the next major version.
+Regardless of the terms used, with a deprecation policy:
+
+1. The deprecated version/package is still available for installation.
+2. The user is informed of the deprecation and can make an informed decision about continuing to use the package or version.
+3. The build will not break.
+
+## A Comprehensive Policy
```

Comment:
Some other ideas we can consider here: 

Have a public list of deleted packages allowing auditors, tooling, proxies or replicas to react or propagate the deletion. I believe GH has/had a public DMCA list that i think we all should respect 🙂 (it is inspirational).

Recommend having deletion policies qualified on illegal versus malicious versus author request. At least for NuGet some of the considerations go away if the content is illegal. We will delete popular illegal content.